Update configuration-settings.rst (#5270)
mattermost/docs: source/configure/configuration-settings.rst

Teams and user accounts can only be created by a verified email from this list of comma-separated domains (e.g. "corp.mattermost.com, mattermost.org").

> This setting only affects email login. For domains to be restricted, you must also set Require Email Verification to ``True``. 

Reviewer: ```suggestion
This setting only affects email login. For domains to be restricted, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 
```

Reviewer: Added a direct link to the config setting

> This setting only affects email login. For domains to be restricted, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 

Reviewer 2: ```suggestion
This setting only affects email login. For domains restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 
```

> This setting only affects email login. For domains restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``. 

Reviewer 3: ```suggestion
This setting only affects email login. For domain restrictions to be effective, you must also set `Require Email Verification <https://docs.mattermost.com/configure/configuration-settings.html#require-email-verification>`__ to ``true``.
```
